valhalla/config.h

```diff
@@ -31,3 +37,57 @@
 
 /* Define to the version of this package. */
 #define PACKAGE_VERSION VALHALLA_VERSION
+
+namespace valhalla {
+const boost::property_tree::ptree& config(const std::string&);
+
+class ConfigUninitializedException : std::exception {
+public:
+  const char* what() const throw() {
+    return "Config singleton was not initialized before usage";
+  }
+};
+} // namespace valhalla
+
+namespace {
+struct config_singleton_t {
+protected:
+  boost::property_tree::ptree config_;
+
+  config_singleton_t() = delete;
+  config_singleton_t(const std::string& config_file_or_inline) {
+    if (config_file_or_inline.empty()) {
+      throw valhalla::ConfigUninitializedException();
```

Comment:
why not just use `throw std::runtime_error("Config singleton was not initialized before usage");`? it must be because we have try catch blocks in things like the workers which key off the type of exception? perhaps the workers have a new catch clause that returns a specific error when not configured? or maybe it should be a valhalla_exception_t with a specific error code? fwiw this is fine, dont feel the need to change it, i more just wanted to discuss it/think about it a little.